pandas/core/indexes/range.py

```diff
@@ -164,7 +164,7 @@ def _simple_new(cls, values: range, name: Hashable = None) -> RangeIndex:
         assert isinstance(values, range)
```

Comment:
@simonjayhawkins are the mypy checks thorough/reliable enough that we can safely remove these assertions?